File: exploreArrays/C++/Ejercicio6.cpp
#include <vector>
#include <iostream>
using namespace std;
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int length = 0;
        for(int i = 0; i< nums.size(); i++){
            length++;
        }
        for(int i = 0; i < length; i++){
            while(nums[i] == val && i < length){
                for(int j = i + 1; j < length; j++){
                    nums[j - 1] = nums[j];
                }
                length--;
            }
        }
        return length;
    }
};
int main(){
    vector<int> vec = {0, 1, 2, 2, 3, 0, 4, 2};
    Solution sol;
    cout << sol.removeElement(vec, 2);
    for(int i = 0; i< vec.size(); i++){
        cout << vec[i];
    }    
}

File: exploreHashmaps/Python/Ejercicio6.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1:
            if n in seen:
                return False
            seen.add(n)
            sq_sum = 0
            while n > 0:
                sq_sum += (n%10)**2
                n = n//10
            n = sq_sum
        return True
sol = Solution()
#print(sol.isHappy(19))
print(sol.isHappy(20))
print(sol.isHappy(68))
#print(sol.isHappy(100))
            
            